Topics/System_Design/Intro_To_Microservices/Intro_To_Microservices.md

```diff
@@ -0,0 +1,165 @@
+# Introduction to Microservices Architecture
+
+## Table of Contents
+
+1. [Overview of Microservices](#overview)
+2. [Advantages and Disadvantages](#adv-dis)
+4. [Principles and Characteristics](#principles)
+5. [Design Considerations](#design)
+6. [Tools and Technologies](#common-tech)
+7. [Best Practices](#best-practices)
+8. [Real-World Examples](#examples)
+
+
+
+## 1. Overview of Microservices <a name="overview"></a>
+
+In the realm of CSC301H1: Introduction to Software Engineering, understanding modern software architecture is crucial. Microservices architecture, a cornerstone of contemporary software development, is a concept that aligns seamlessly with the principles taught in this course.
+
+Microservices architecture presents a paradigm shift in how software systems are designed and developed. Unlike traditional monolithic architectures, where applications are built as a single, tightly integrated unit, microservices architecture emphasizes breaking down complex systems into smaller, independently deployable services.
```

Comment:
This is a good definition but I think you can elaborate a little more on monolithic architecture vs microservices architecture. Maybe talk a little more about the differences.

Reply:
For sure, I will add a brief line about that but I believe there is a dedicated section for that.